src/main/java/thewaypointers/trafficsimulator/gui/ControlPanel.java

```diff
@@ -42,6 +46,8 @@
 
     Graphics2D g2;
 
+    final TrafficSimulatorManager trafficSimulatorManager = new TrafficSimulatorManager(startPauseButton);
```

Comment:
ControlPanel shouldn't create TrafficSimulatorManager, that's unnecessary dependency. It should receive it from outside (in the constructor for example).